### PR TITLE
testing/py-ansible-lint: upgrade, add check, remove broken subpackage

### PR DESCRIPTION
**Pull request by maxwell-k**
This pull request addresses issues reported at: https://bugs.alpinelinux.org/issues/9145
It proposes four distinct changes:

- upgrade to the most recent release 3.4.23
- explicitly list python package dependencies not shared with Ansible - six
- add a basic `check()`
- remove the non-functional `py2-` sub package

### CODE REVIEW

**Comment by maxwell-k** [comment]
@fabaff are you the maintainer of `ansible-lint` in Alpine Linux? Could you please take a look at this? I am happy to incorporate any feedback.

**Comment by andypost** [comment]
Itblooks like it should be 3 commits: upgrade,modernize, rename

**Comment by maxwell-k** [comment]
@andypost thanks for the feedback; I have incorporated your advice. 

Any other changes needed?

**Comment by andypost** [comment]
it looks good but I guess when package moved it needs to add BC kinda
```
provides="py3-ansible-lint=$pkgver-r$pkgrel" # for backward compatibility
replaces="py3-ansible-lint" # for backward compatibility
```
But better to wait maintainer's opinion /cc @fabaff

**Comment by maxwell-k** [comment]
No opinion so far (4 weeks) so I have incorporated your suggestion @andypost and kept it to two commits.

File: testing/ansible-lint/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Contributor: Fabian Affolter <fabian@affolter-engineering.ch>
+# Maintainer: Fabian Affolter <fabian@affolter-engineering.ch>
+pkgname=ansible-lint
+pkgver=3.4.23
+pkgrel=0
+pkgdesc="A tool to check ansible playbooks"
+url="https://github.com/willthames/ansible-lint"
+arch="noarch"
+license="MIT"
+depends="python3 ansible py3-setuptools py3-six py3-yaml"
```
Comment (fabaff) [inline testing/ansible-lint/APKBUILD]: `ansible` is already pulling in `py3-yaml`.

**Comment by maxwell-k** [comment]
Since I started this PR `ansible-lint` has been adopted as an official Red Hat project, so the first commit now shows an updated URL.
https://groups.google.com/forum/#!topic/ansible-project/ehrb6AEptzA

File: testing/ansible-lint/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Contributor: Fabian Affolter <fabian@affolter-engineering.ch>
+# Maintainer: Fabian Affolter <fabian@affolter-engineering.ch>
+pkgname=ansible-lint
+pkgver=3.4.23
+pkgrel=0
+pkgdesc="A tool to check ansible playbooks"
+url="https://github.com/ansible/ansible-lint"
+arch="noarch"
+license="MIT"
+depends="python3 ansible py3-setuptools py3-six"
```
Comment (andypost) [inline testing/ansible-lint/APKBUILD]: I bet this should be just `depends="ansible py3-six"`
and next line `makedepends="python3-dev py3-setuptools"`

Comment (andypost) [reply testing/ansible-lint/APKBUILD]: btw There's issue about to split https://bugs.alpinelinux.org/issues/9622

Comment (maxwell-k) [reply testing/ansible-lint/APKBUILD]: OK understand, see the benefit in moving setuptools to makedepends and get that python3 was redundant if ansible is listed just like py3-yaml was in fabaff's review. Thanks for your feedback!

I'm going to watch that bug about the split

**Comment by andypost** [comment]
@fabaff Maybe move it to community instead of testing?

**Pull request closed without merging**